src/super_gradients/common/auto_logging/console_logging.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from io import StringIO
+import atexit
+
+from super_gradients.common.environment.env_helpers import multi_process_safe, is_distributed
+
+
+FILE_BUFFER_SIZE = 10_000
+
+
+class BufferWriter:
+    """File writer buffer that opens a file only when flushing and under the condition that threshold buffersize was reached."""
+
+    def __init__(self, filename: str, buffer_size: int):
+        self.buffer = StringIO()
+        self.filename = filename
+        self.buffer_size = buffer_size
+        self.open = open
+
+    def write(self, data: str):
+        self.buffer.write(data)
```

Comment:
if you have a defined buffer size, the point is to check if a flush is required after each write.
so you accumulate the strings into the buffer, but every once in a while it is written to file

Reply:
I changed it like that: https://github.com/Deci-AI/super-gradients/blob/8bfd183beb1ba34012bd58dd0ae3c3a91cf164c1/src/super_gradients/common/auto_logging/console_logging.py#L22-L38

Reply:
looks ok